File: ProjectEuler_FSharp/EulerProblem3.fs
module ProjectEuler_FSharp.EulerProblem3

open System

let Calculate(maxValue : int64) =
    let valRoot = int64 (Math.Sqrt(float maxValue)) + 10L
    Primes.UpToBig valRoot |> Seq.rev |> Seq.pick (fun p -> if maxValue % p = 0L then Some(p) else None)

File: ProjectEuler_CSharp/EulerProblem3.cs
using System;
using System.Linq;

namespace ProjectEuler_CSharp
{
    public static class EulerProblem3
    {
        public static long Calculate(long value)
        {
            foreach (var i in Primes.UpTo((int)Math.Sqrt(value) + 10).Reverse())
                if (value % i == 0)
                    return i;
            throw new Exception($"{value} is prime");
        }
    }
}
